frontend/src/metabase/common/components/EntityPicker/hooks/use-scoped-search-results.ts

```diff
@@ -44,7 +74,12 @@ export const useScopedSearchResults = <
 
   const { data: collectionItemsData, isFetching: isFetchingCollectionItems } =
     useListCollectionItemsQuery(
-      shouldUseCollectionItems ? { id: folder.id as CollectionId } : skipToken,
+      shouldUseCollectionItems
+        ? {
+            id: folder.id as CollectionId,
+            models: searchModelsToCollectionItemModels(searchModels),
```

Comment:
why not just ?
```suggestion
            models: searchModels as as CollectionItemModel[],
```

are there other values that we're filtering out? if we really want to check something at runtime, let's make a type guard

Reply:
Yeah we have to filter out 'tables' which is passed in here sometimes.

Reply:
or at least that the value passed in one test I wanted to be sure there weren't others so I filtered to just the types that are valid here.

Reply:
in that case I'd probably do something like

```suggestion
const searchModelsToCollectionItemModels = (
  searchModels: string[],
): CollectionItemModel[] => {

  const validModels = [
    "card",
    "collection",
    "dataset",
    "dashboard",
    "snippet",
    "metric",
  ] as const;

  return validModels.filter(
    (model) => searchModels.includes(model),
  );
}
```